lib/ui/page/buy.dart

```diff
@@ -0,0 +1,437 @@
+    assetId ??= erc20USDT;
+
+    final supportedAssets = useMemoizedFuture(
+        () => context.appServices.findOrSyncAssets(supportedCryptosId)).data;
+    if (supportedAssets == null || supportedAssets.isEmpty) {
+      w('supportedAssets: $supportedAssets');
+      return const SizedBox();
+    }
+
+    final fiatList = useMemoized<List<WyreFiat>>(getWyreFiatList);
+    return _Buy(supportedAssets: supportedAssets, fiatList: fiatList);
+  }
+}
+
+class _Buy extends HookWidget {
+  _Buy({
+    Key? key,
+    required this.supportedAssets,
+    required this.fiatList,
+  }) : super(key: key);
+
+  final List<AssetResult> supportedAssets;
+  final List<WyreFiat> fiatList;
+
+  final _debouncer = Debouncer(const Duration(milliseconds: 500));
+
+  @override
+  Widget build(BuildContext context) {
+    final asset = useState(supportedAssets[0]);
+    final fiat = useState(fiatList[0]);
+    final type = useState(WyrePayType.applePay);
+    final wyreQuote = useState<WyreQuote?>(null);
+    final lastQuoteByFiat = useState(true);
+
+    final fiatController = useTextEditingController();
+    final cryptoController = useTextEditingController();
+    final fiatFocusNode = useFocusNode(debugLabel: 'fiat input');
+    final cryptoFocusNode = useFocusNode(debugLabel: 'crypto input');
+
+    final country = useMemoized(getCountry);
+
+    void showAssetListBottomSheet() {
+      showMixinBottomSheet(
+        context: context,
+        isScrollControlled: true,
+        builder: (context) => AssetSelectionListWidget(
+          onTap: (AssetResult assetResult) {
+            context.replace(buyPath.toUri({'id': assetResult.assetId}));
+            asset.value = assetResult;
+          },
+          selectedAssetId: asset.value.assetId,
+          assetResultList: supportedAssets,
+        ),
+      );
+    }
+
+    void showFiatListBottomSheet() {
+      showMixinBottomSheet(
+        context: context,
+        isScrollControlled: true,
+        builder: (context) => FiatSelectionListWidget(
+          fiatList: fiatList,
+          selectedFiat: fiat.value,
+          onTap: (WyreFiat wyreFiat) {
+            fiat.value = wyreFiat;
+          },
+        ),
+      );
+    }
+
+    Future<WyreQuote> queryOrderQuote(bool byFiat) async {
+      final data = {
+        'sourceCurrency': fiat.value.name,
+        'destCurrency': asset.value.symbol,
+        'dest': 'ethereum:${asset.value.destination}',
+        'country': country,
+        'accountId': Env.wyreAccount,
+        'walletType': type.value.forQuote(),
+        'amountIncludeFees': 'true',
+      };
+      if (byFiat) {
+        data['sourceAmount'] = fiatController.text;
+      } else {
+        data['destAmount'] = cryptoController.text;
+      }
+      final quote =
+          await WyreClient.instance.api.getOrderReservationQuote(data);
+      return quote;
+    }
+
+    useEffect(() {
+      void updateAmount() {
+        _debouncer.run(() async {
```

Comment:
Debouncer looks a bit complex, Would you make TextEditingController convert to Stream, and use it with Rx’s extension method debounceTime?

Reply:
try refactor using stream